recursion.rb

```diff
@@ -1,16 +1,71 @@
 
+
+  # RC: n! = n * ( n - 1 )!
+  # BC: 1! = 1
+  # BC: 0! = 1
+
 def fact(n)
+  #BC:
+  if n == 0 || n == 1
+    return 1
+  end
+  #RC:
+  return n * fact(n-1)
 end
 
+# adds the number before it:
+# 1 1 2 3 5 8 13 24
+# fib(4) = 3
+# fib(1) = 1
+# fib(2) = 1
+# fib(3) = fib(2) + fib(1)
+# fib(4) = fib(3) + fib(2)
+# fib(6) = fib(5) + fib(4)
+
 def fib(n)
+  #BC:
+  if n == 1 || n == 2
+    return 1
+  end
+  #RC:
+    return fib(n-1) + fib(n-2)
 end
 
+
+# do this for homework!
 def pal(s)
+  i = 0
+  s.chars.length
+  while i <= s / 2
+    if s[i] != s[length - 1 - i]
+      return false
+        i += 1
+      return true
+    end
+  end
 end
 
+
+
 def binary(n)
+  binary_helper(n, "", "")
+end
+
+def binary_helper(digits, binary, result)
+  #BC:
+  if digits == 0
+    result += binary + " "
+  else
+    #RC:
+    result = binary_helper(digits - 1, binary + "0", result)
+    result = binary_helper(digits - 1, binary + "1", result)
+  end
+  return result
 end
```

Comment:
Thanks for checking out my code!

On Sun, May 15, 2016 at 9:55 PM, malderi <notifications@github.com> wrote:

> In recursion.rb
> <https://github.com/Ada-C5/recursion/pull/21#discussion_r63309933>:
>
> >  def binary(n)
> > +  binary_helper(n, "", "")
> > +end
> > +
> > +def binary_helper(digits, binary, result)
> > +  #BC:
> > +  if digits == 0
> > +    result += binary + " "
> > +  else
> > +    #RC:
> > +    result = binary_helper(digits - 1, binary + "0", result)
> > +    result = binary_helper(digits - 1, binary + "1", result)
> > +  end
> > +  return result
> >  end
>
> Good solution!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Ada-C5/recursion/pull/21/files/eaa162b4ad6cfef68b017cdbb164d55ccc3d208e#r63309933>
>